### PR TITLE
KEYCLOAK-7635 Authenticate clients with x509 certificate

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/KEYCLOAK-7635

This PR contains documentation for https://github.com/keycloak/keycloak/pull/5435. Please make sure it is merged after the one against Keycloak code.

### CODE REVIEW

**Review by sebastienblanc — DISMISSED** [review]
:+1:

**Comment by slaskawi** [comment]
@matthewhelmke Unfortunately I can't access the logs and can't figure out what's wrong with this commit. May I ask you for help?

**Comment by matthewhelmke** [comment]
@slaskawi I took a look. There are some validation failures, but all are known issues and marked with "IGNORING: Validation is broken, see BZ #1097495". You are fine. The actual build is error-free and successful.

**Review by sebastienblanc — DISMISSED** [review]

**Comment by slaskawi** [comment]
@sebastienblanc Added also release notes.

**Review by matthewhelmke — DISMISSED** [review]
The invalid links issue should have been fixed in https://github.com/keycloak/keycloak-documentation/pull/445. I think this is good to go.

File: release_notes/topics/4_1_0_final.adoc [hunk 1]
```diff
@@ -1,3 +1,8 @@
 = Making Spring Boot 2 the default starter
 
-Starting with release 4.1, the Spring Boot starter will be based on the Spring Boot 2 adapter. If you are using an older Spring Boot version, the keycloak-legacy-spring-boot-starter is available.
\ No newline at end of file
+Starting with release 4.1, the Spring Boot starter will be based on the Spring Boot 2 adapter. If you are using an older Spring Boot version, the keycloak-legacy-spring-boot-starter is available.
```
Comment (stianst) [inline release_notes/topics/4_1_0_final.adoc]: This is the wrong place. You're adding to 4.1 release notes, while we're doing 4.3.

**Review by stianst — CHANGES_REQUESTED** [review]

File: release_notes/topics/4_1_0_final.adoc (hunk 1 above)
Comment (sebastienblanc) [reply release_notes/topics/4_1_0_final.adoc]: Actually it's in since 4.2 but I wonder why this change is in this PR.

Comment (stianst) [reply release_notes/topics/4_1_0_final.adoc]: Wasn't there some additions added after 4.2? For the subject dn checks?

**Review by stianst — APPROVED** [review]

**Pull request merged**